Frends.SFTP.DownloadFiles/Frends.SFTP.DownloadFiles/Definitions/FileTransporter.cs

```diff
@@ -73,39 +73,73 @@ public FileTransferResult Run(CancellationToken cancellationToken)
                 client.ConnectionInfo.KeyExchangeAlgorithms.Remove("curve25519-sha256");
                 client.ConnectionInfo.KeyExchangeAlgorithms.Remove("curve25519-sha256@libssh.org");
 
+                var expectedServerFingerprint = _batchContext.Connection.ServerFingerPrint;
                 // Check the fingerprint of the server if given.
-                if (!String.IsNullOrEmpty(_batchContext.Connection.ServerFingerPrint))
+                if (!string.IsNullOrEmpty(expectedServerFingerprint))
                 {
+                    SetCurrentState(TransferState.Connection, "Checking server fingerprint.");
                     try
                     {
-                        SetCurrentState(TransferState.Connection, "Checking server fingerprint.");
-                        // If this check fails then SSH.NET will throw an SshConnectionException - with a message of "Key exchange negotiation failed".
+                        #region HostKeyAuthentication
```

Comment:
Instead of doing a region here and creating a delegate (for HostKeyReceived event) - just extract this whole cert verification thing into its own method. Otherwise this is a very long and complex method.

Reply:
Extracted this to a method of its own.